### ISSUE TITLE
Add HOWTO: Implement beam search

### ISSUE THREAD

**Issue by marcvanzee** [google/flax#875]
We currently have implemented this in our WMT example in [decode.py](https://github.com/google/flax/blob/master/examples/wmt/decode.py). It presumes a function that takes integer tokens and returns logits. It has been optimized for TPUs.
